### PR TITLE
Use folium of the same folder instead of installed version when generating the docs

### LINKED ISSUE #983: Creating documentation latex files fails (Python 3.5)
#### Please add a code sample or a nbviewer link, copy-pastable if possible

```shell
cd ./docs
make latex
```

#### Problem description

Instead of creating the tex files the following happens:

```shell
sphinx-build -b latex -d _build/doctrees   . _build/latex
Running Sphinx v1.8.1

Configuration error:
There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "/XXXXXXXXX/.local/lib/python3.5/site-packages/sphinx/config.py", line 368, in eval_config_file
    execfile_(filename, namespace)
  File "/XXXXXXXXX/.local/lib/python3.5/site-packages/sphinx/util/pycompat.py", line 150, in execfile_
    exec_(code, _globals)
  File "/YYYYYYYYY/folium/folium/docs/conf.py", line 52, in <module>
    from folium._version import get_versions
ImportError: No module named 'folium'

Makefile:99: recipe for target 'latex' failed
make: *** [latex] Error 2
```

#### Expected Output

Nice latex files which I could use for creating a PDF.

#### Output of ``folium.__version__``
     0.6.0+30.g9f648a2.dirty'

### PR DESCRIPTION
**Pull request by 1kastner**
Closes #983.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -11,7 +11,12 @@
 # All configuration values have a default; values that are commented out
 # serve to show the default.
 
-import sys, os
+import sys
+import os
+import os.path
```
Comment (Conengmo) [inline docs/conf.py]: Importing `os.path` is not needed

File: docs/conf.py [hunk 2]
```diff
@@ -11,7 +11,12 @@
...
+
+# Use the currently checked out folium of this folder
+sys.path.insert(0, os.path.join(os.pardir))
```
Comment (Conengmo) [inline docs/conf.py]: `os.path.join()` is doing nothing here. Did you mean `os.path.abspath()`?

Comment (1kastner) [reply docs/conf.py]: No, I was just sleeping. It can be removed.

**Comment by Conengmo** [comment]
Do we need to make the path explicit with `os.path.abspath` or is the current relative path sufficient?

**Comment by 1kastner** [comment]
@Conengmo that's indeed a good question. Maybe we should try it out on a fresh machine without folium installed. Then for the relative path it should fail to import in case it is necessary.

**Comment by Conengmo** [comment]
If you're not sure either I would suggest to put it in. I've had unexpected issues before with relative paths in CI environments. But if you want to test it out that's fine as well.

**Comment by 1kastner** [comment]
Let's be on the safe side

**Comment by Conengmo** [comment]
It seems autobuilding the docs went fine. Thanks @1kastner!

**Pull request merged**